fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/mixin/tool/attribute/MixinItemStack.java

```diff
@@ -42,30 +41,12 @@
 
 	@Inject(at = @At("HEAD"), method = "isEffectiveOn", cancellable = true)
 	public void isEffectiveOn(BlockState state, CallbackInfoReturnable<Boolean> info) {
-		TriState triState = ToolManager.handleIsEffectiveOn((ItemStack) (Object) this, state, null);
-
-		if (triState != TriState.DEFAULT) {
-			info.setReturnValue(triState.get());
-			info.cancel();
-		}
+		info.setReturnValue(ToolManager.handleIsEffectiveOn(state, (ItemStack) (Object) this, null).get());
```

Comment:
I horribly dislike the fact that we are using a short circuiting injector to handle this. Other mods which mixin to `isEffectiveOn` will silent fail if they inject after fabric here.

I'd say `@Overwrite` or find a way to keep the vanilla codepath within a possible scope to be executed.